api/v1alpha1/managedcertificate_types.go

```diff
@@ -0,0 +1,109 @@
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// Important: Run "make crd" to regenerate code after modifying this file
+// NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
+
+// ManagedCertificateSpec defines the desired state of ManagedCertificate.
+type ManagedCertificateSpec struct {
+	// Target defines the targets that the Certificate will be bound to.
+	// +kubebuilder:validation:Required
+	Target ManagedCertificateTarget `json:"target,omitempty"`
+
+	// DnsZone defines the DNS Zone that the ManagedCertificate will be applied to.
+	// +kubebuilder:validation:Required
+	DnsZone ManagedCertificateDnsZone `json:"dnsZone,omitempty"`
+
+	// DomainNames is a list of domain names that the Certificate will be issued for.
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:MinItems=1
+	// +listType=set
+	DomainNames []string `json:"domainNames,omitempty"`
+
+	// ServiceAccount is the name of the ServiceAccount that will be used to connect to the Azure DNS Zone.
+	// +kubebuilder:validation:Required
+	ServiceAccount string `json:"serviceAccount,omitempty"`
+}
+
+// ManagedCertificateTarget defines the targets that a Certificate will be bound to.
+// +kubebuilder:validation:MinProperties=1
+// +kubebuilder:validation:MaxProperties=1
+type ManagedCertificateTarget struct {
+	// Secret is the name of the Secret that will contain the Certificate.
+	// +kubebuilder:validation:Required
+	Secret string `json:"secret,omitempty"`
+}
+
+// ManagedCertificateDnsZone defines the DNS Zone that a ManagedCertificate will be applied to.
+type ManagedCertificateDnsZone struct {
+	// ResourceId is the Azure Resource ID of the DNS Zone. Can be retrieved with `az network dns zone show -g <resource-group> -n <zone-name> --query id -o tsv`.
+	ResourceId string `json:"resourceId,omitempty"`
+
+	// below fields are needed for cross-tenant scenarios
+
+	// TenantId is the Azure Tenant ID of the DNS Zone.
+	// +kubebuilder:validation:Optional
+	TenantId string `json:"tenantId,omitempty"`
```

Comment:
behavior on DNS CRDs is to have tenant ID as a top level, required field. Not saying one is correct vs the other, but we should likely have them match. Do you think we should have this be required, or have it be optional in both and use config's tenantID?

Reply:
it should default to the cluster's tenant in both

Reply:
most users shouldn't ever need this field

Reply:
alright, I'll put together a PR changing that for existing DNS resources once clusterDNS controller PR is merged